cpg-neo4j/src/main/java/de/fraunhofer/aisec/cpg_vis_neo4j/Application.kt

```diff
@@ -243,16 +268,60 @@ class Application : Callable<Int> {
             }
             filePaths[index] = file
         }
+        if (exclusive.jsonCompilationDatabase == "" && exclusive.files.isEmpty()) {
+            throw Error(
+                "Files list is empty or jsonCompilationDatabase is also empty. Please provide --files to evaluate or --json-compilation-database for CXX files"
+            )
+        }
 
         val translationConfiguration =
             TranslationConfiguration.builder()
                 .sourceLocations(*filePaths)
-                .topLevel(topLevel!!)
+                .topLevel(topLevel)
                 .defaultPasses()
                 .defaultLanguages()
                 .loadIncludes(loadIncludes)
                 .debugParser(DEBUG_PARSER)
 
+        if (exclusive.jsonCompilationDatabase != "") {
+            val jsonStringFile = File(exclusive.jsonCompilationDatabase).readText()
+            val mapper = ObjectMapper().registerKotlinModule()
+            val obj: List<CompilationDbStructure> = mapper.readValue(jsonStringFile)
```

Comment:
I would move this functionality into the CXX language frontend, so that we can use it independently of cpg-neo4j

Reply:
Added this